feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+	}
+
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cpuCards := components.FindComponentsByType(t, dut, cpuType)
+	for _, cpu := range cpuCards {
+		timestamp := time.Now().Round(time.Second)
+
+		query := gnmi.OC().Component(cpu).State()
+		component := gnmi.Get(t, dut, query)
+
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		// If cpu card's parent is line card, check cpu ultilization.
+		if slices.Contains(lineCards, cpuParent) {
+			// Remove parent from the list of check cards.
+			lineCards = removeElement(lineCards, cpuParent)
+			// Fetch CPU utilization data.
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization.Avg == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%", timestamp, deviceName, cpu, cpuParent, cpuUtilization.GetAvg())
+		}
+	}
+
+	if len(lineCards) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", lineCards)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+	description := "Component"
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		if componentType == lineCardType && deviations.LinecardMemoryUtilizationUnsupported(dut) {
+			t.Logf("INFO: Skipping test for linecard component %s due to deviation linecard_memory_utilization_unsupported", component)
+			continue
+		}
+
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%", timestamp, deviceName, description, component, componentType, memoryUtilization)
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	const description = "System CPU Process"
+
+	timestamp := time.Now().Round(time.Second)
+	results := gnmi.GetAll(t, dut, query)
+	for _, result := range results {
```

Comment:
t.Run() 

And add a failing case for no processes scenario?

Reply:
Thanks for your comments. 
1. Add t.Run()
```go
	results := gnmi.GetAll(t, dut, query)
	for _, result := range results {
		processName := result.GetName()
		t.Run(processName, func(t *testing.T) {
```
2. Won't GetAll cause script fail if there is no processes?

Reply:
1: looks good.

2:  AFAIK `GetAll` will not fail if the leaf/container is supported but there are no processes. @wenovus can you confirm?

Reply:
@DanG100 confirmed that `GetAll()` will fail if there are no processes. So no need to add additional condition for that case.